core-build/gulp-core-build-sass/src/CSSModules.ts

```diff
@@ -0,0 +1,57 @@
+import * as path from 'path';
+
+import * as postcss from 'postcss';
+import * as cssModules from 'postcss-modules';
+import * as crypto from 'crypto';
+
+export interface ICSSModules {
+  /**
+   * Return a configured postcss plugin that will map class names to a
+   * consistently generated scoped name.
+   */
+  getPlugin: () => postcss.AcceptedPlugin;
+
+  /**
+   * Return the CSS class map that is stored after postcss-modules runs.
+   */
+  getCssJSON: () => Object;
+}
+
+export default class CSSModules implements ICSSModules {
+  private _classMap: Object;
+  private _rootPath: string;
+
+  constructor(rootPath?: string) {
+    this._classMap = {};
+    if (rootPath) {
+      this._rootPath = rootPath;
+    } else {
+      this._rootPath = process.cwd();
+    }
+  }
+
+  public getPlugin = () => {
```

Comment:
Can these be conventional member functions?

Reply:
THAT is an excellent question, and one that I would like to know. Back in the early ES6 days, member functions always "properly" mapped `this` to the object instance like I thought they should (according to my recollection). Then at some point  (ES7 or ES8 maybe?) that changed and you had to use some syntax that I found rather awkward (`this.memberFunc.bind(this)`). That's when I found out from another dev that arrow functions in the latest incarnation would auto bind to the instance `this` like how the old member functions used to. But, I'm not an authority on the topic, or how TypeScript may interpret member functions, or even what version of ES our tooling supports. Therefore, I would like to know the answer to that question too. (Honestly, I find the arrow function syntax awkward as well, but slightly less so than always having to call `.bind(this)`).

Reply:
Or were you just asking about the public functions? I think those do **not** need the arrow syntax, but the `saveJson` and `generateScopedName` methods either need arrow syntax or `.bind(this)` later on. And I just kept them all the same for conformity's sake.

Reply:
I spent hours reading up on this, including changes coming in new ECMAScript specs. Apparently, even the language authors debate this. So I just reverted everything to using member functions for the sake of consistency. Theoretically, this is slightly more performant  too.